src/packages/core/property-editor/uis/tree-picker/config/filter/property-editor-ui-tree-picker-filter.element.ts

```diff
@@ -0,0 +1,95 @@
+import { UMB_DATA_TYPE_WORKSPACE_CONTEXT } from '../../../../../data-type/workspace/data-type-workspace.context.js';
+import { UmbDocumentTypeInputElement } from '../../../../../../documents/document-types/components/document-type-input/document-type-input.element.js';
+import { UmbMediaTypeInputElement } from '../../../../../../media/media-types/components/media-type-input/media-type-input.element.js';
+import { StartNode } from '@umbraco-cms/backoffice/components';
+import type { UmbPropertyEditorUiElement } from '@umbraco-cms/backoffice/extension-registry';
+import { customElement, html, property } from '@umbraco-cms/backoffice/external/lit';
+import { UmbTextStyles } from '@umbraco-cms/backoffice/style';
+import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
+
+@customElement('umb-property-editor-ui-tree-picker-filter')
+export class UmbPropertyEditorUITreePickerFilterElement extends UmbLitElement implements UmbPropertyEditorUiElement {
+	constructor() {
+		super();
+
+		this.consumeContext(UMB_DATA_TYPE_WORKSPACE_CONTEXT, (instance) => {
+			const workspace = instance;
+			this.observe(workspace.data, (data) => {
```

Comment:
Hi @leekelleher :-)

In this case you can use: `workspace.propertyValueByAlias()` and in that way only listen for that specific value. This will lower the amount of executions on this callback. In other words be more performant.

Ideally we always provide Observables that can be specific to most use cases.

Just a bit of background: :-)
In above case the `.data` property is all data, and will be triggered every time there is one change to it.
And then we have several Observables ( offen made via a helper method called .asObservablePart(...) ) which then only triggers if that specific part changes.

Reply:
@nielslyngsoe Thanks for the tip, I'll try to implement it that way.  Do you know of any existing examples in the codebase? as I haven't been able to find any (yet).

With my initial attempt of observing the entire data-type workspace, I'd copied similar code from the `umb-multiple-color-picker-input`, e.g. Approved Color editor's "Include labels?" toggle.
https://github.com/umbraco/Umbraco.CMS.Backoffice/blob/28f8de57a5833a9a15ded74816d6314ac92be20e/src/packages/core/components/multiple-color-picker-input/multiple-color-picker-input.element.ts#L116-L122

Once I have a working example of observing the `workspace.propertyValueByAlias()`, then we can also update the Approved Color editor configuration accordingly.

Reply:
@nielslyngsoe I've figured out the code, see commit 4f8739294c8f9e6aa2beafdd2336e8b434ba323b for the snippet.